tests/src/test/scala/it/unibo/scafi/test/functional/stdlib/TestBlockT.scala

```diff
@@ -0,0 +1,166 @@
+package it.unibo.scafi.test.functional.stdlib
+
+import it.unibo.scafi.test.FunctionalTestIncarnation._
+import it.unibo.scafi.test.functional.ScafiAssertions.assertNetworkValues
+import it.unibo.scafi.test.functional.ScafiTestUtils
+import it.unibo.utils.StatisticsUtils._
+import org.scalatest._
+
+
+/*
+Not yet tested:
+- cyclicTimerWithDecay
+- clock
+- impulsesEvery
+ */
+class TestBlockT extends FlatSpec{
+  import ScafiTestUtils._
+
+  val Block_T = new ItWord
+
+  private[this] trait SimulationContextFixture {
+    val net: Network with SimulatorOps = ScafiTestUtils.standardNetwork()
+  }
+
+  private[this] trait TestProgram extends AggregateProgram with StandardSensors with BuildingBlocks
+
+  def unitaryDecay: Int => Int = _ - 1
+  def halving: Int => Int = _ / 2
+
+  Block_T should "support T with unitary decay and 0 floor value" in new SimulationContextFixture {
+    exec(new TestProgram {
+      override def main(): Int = T(10, 0, unitaryDecay)
+    }, ntimes = someRounds)(net)
+
+    assertNetworkValues((0 to 8).zip(List(
+      0, 0, 0,
+      0, 0, 0,
+      0, 0, 0
+    )).toMap)(net)
+  }
+
+  Block_T should "T should initialize as specified" in new SimulationContextFixture {
+    exec(new TestProgram {
+      override def main(): Int = T(10, 0, identity[Int])
+    }, ntimes = fewRounds)(net)
+
+    assertNetworkValues((0 to 8).zip(List(
+      10, 10, 10,
+      10, 10, 10,
+      10, 10, 10
+    )).toMap)(net)
+  }
+
+  Block_T should "T should be consistent in intermediate tests" in new SimulationContextFixture {
+    exec(new TestProgram {
+      override def main(): (Int, Int) = (
+        T(manyManyRounds, 0, unitaryDecay),
+        //this can be seen as a round counter
+        rep(0)(_ + 1)
+      )
+    }, ntimes = fewRounds)(net)
+
+    assert(net.valueMap[(Int, Int)]().forall { case (_, (done: Int, todo: Int)) => done + todo == manyManyRounds })
```

Comment:
Aren't `done` and `todo` inverted?
In any case, I'd change their name to clarify what they are, i.e., `decayTimer` and `roundsPerformed`.
The invariant should be that `decayTimer+roundsPerformed >= ceiling+1`, right?

However, this does not really test intermediate states.
I was thinking that maybe we could perform assertions round by round, and check e.g. that `decayTimer` actually decays (by comparing wrt the previous value, which we may store in a var). We could actually use a tiny network made of two devices, as no interaction happens with `T`.

Reply:
They are inverted :/
The idea behind this test is to be sure that T won't be able to complete, in this way the assertion is performed in an intermediate state